src/Compiler/Scheme/Common.idr

```diff
@@ -24,6 +25,37 @@ schString s = concatMap okchar (unpack s)
                   then cast c
                   else "C-" ++ show (cast {to=Int} c)
 
+export
+startSchemePreamble : String -> String -> String
+startSchemePreamble appdir backend = """
+  #!/bin/sh
+  # \{ generatedString backend }
+
+  set -e # exit on any error
+
+  case "$(uname -s)" in
+        Linux*)     OS=Linux;;
+        Darwin*)    OS=Mac;;
+        CYGWIN*)    OS=Windows;;
+        MINGW*)     OS=Windows;;
```

Comment:
This won't be used on Windows unless the call to startChezWinSh in makeShWindows is replaced.

Reply:
unfortunately i have little understanding of cygwin and windows, and no easy way to test on windows, so i have left it alone as-is. it was quite a headache already to make this work on windows without a local dev env.

would it be possible that someone else incorporates this in a post-PR commit? or maybe even push a commit into this PR?

Reply:
or would it just work if i simply deleted `startChezWinSh` and replaced it with a call to `startChez`?

Reply:
It can wait, I just thought that it was intended given the cases for it.

You could try to put it in, if it doesn't work it will be obvious as no tests will run.

Reply:
there's a copy of `startChezWinSh` in `ChezSep.idr`, too. i have no idea what that "sep" is. i'd rather leave it to someone with a windows devenv.